### PR TITLE
Remove UpdateApplication interface from application store

### PR DESCRIPTION
**Pull request by khanhtc1202**
**What this PR does / why we need it**:

In this PR, I removed
- the general use `UpdateApplication` function from ApplicationStore interface.
- the `UpdateApplicationDescription` rpc from web service proto (since we remove that feature on the web console)

For the next step, I want to make separated XXXStore interfaces that be used on the API layer to avoid huge interface issues (the current huge ApplicationStore interface exists because I created a bunch of new functions which are the replacement for UpdateApplication + updater pattern).

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/datastore/applicationstore.go [hunk 1]
```diff
@@ -149,7 +152,7 @@ func (s *applicationStore) DisableApplication(ctx context.Context, id string) er
 	})
 }
 
-func (s *applicationStore) UpdateApplication(ctx context.Context, id string, updater func(*model.Application) error) error {
+func (s *applicationStore) updateApplication(ctx context.Context, id string, updater func(*model.Application) error) error {
```
Comment (nghialv) [inline pkg/datastore/applicationstore.go:155]: `updateApplication` -> `update`

Comment (khanhtc1202) [reply pkg/datastore/applicationstore.go:155]: Nice point 👍

Comment (khanhtc1202) [reply pkg/datastore/applicationstore.go]: updated at https://github.com/pipe-cd/pipecd/pull/3201/commits/839ff7207942440082003d11edc41d974acaf28c 🙏

File: pkg/app/server/grpcapi/api.go [hunk 2]
```diff
@@ -296,9 +296,10 @@ func (a *API) ListApplications(ctx context.Context, req *apiservice.ListApplicat
 		Cursor:  req.Cursor,
 	}
 
-	apps, cursor, err := listApplications(ctx, a.applicationStore, opts, a.logger)
+	apps, cursor, err := a.applicationStore.List(ctx, opts)
 	if err != nil {
-		return nil, err
+		a.logger.Error("failed to list applications", zap.Error(err))
```
Comment (nghialv) [inline pkg/app/server/grpcapi/api.go:301]: nit: We don't need to log this since that kind of log has already been logged at the datastore level.

Comment (khanhtc1202) [inline pkg/app/server/grpcapi/api.go:301]:
```suggestion
```

**Comment by nghialv** [comment]
Nice.
/lgtm

File: pkg/app/server/grpcapi/api.go [hunk 3]
```diff
@@ -314,21 +314,15 @@ func (a *API) RenameApplicationConfigFile(ctx context.Context, req *apiservice.R
...
 		if err != nil {
-			if denied {
-				return nil, status.Error(codes.PermissionDenied, err.Error())
-			}
-			a.logger.Error("failed to update application", zap.Error(err))
-			return nil, status.Error(codes.Internal, fmt.Sprintf("Failed to update application %s", appID))
+			return nil, gRPCEntityOperationError(err, fmt.Sprintf("failed to get application %s", appID))
+		}
+		if app.ProjectId != key.ProjectId {
+			return nil, status.Error(codes.PermissionDenied, fmt.Sprintf("requested application %s does not belong to your project", appID))
```
Comment (knanao) [inline pkg/app/server/grpcapi/api.go:322]: [imo] I think it better that  `PermissionDenied` is also handled by gRPCEntityOperationError. but it's up to you and in another PR.

Comment (knanao) [reply pkg/app/server/grpcapi/api.go:322]: Oh, Let me fix it in another PR! so this PR lgtm!

**Comment by knanao** [comment]
Cool!
/approve

File: pkg/app/server/grpcapi/api.go (hunk 3 above)
Comment (khanhtc1202) [reply pkg/app/server/grpcapi/api.go:322]: Yeh, the current gRPCEntityOperationError does not handle any errors other than datastore's error
https://github.com/pipe-cd/pipecd/blob/master/pkg/app/server/grpcapi/grpcapi.go#L195-L208
So please address what you want by separated PR 🙏

**Pull request merged**